src/main/java/qna/service/QnaService.java

```diff
@@ -35,25 +35,8 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.delete();
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.delete();
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+
+        DeleteHistories delete = question.delete(loginUser);
+        deleteHistoryService.saveAll(delete);
```

Comment:
번거로울 수 있겠지만, 히스토리가 여러개인 의도를 나타내주시면 어떨까요?

```suggestion
        DeleteHistories deleteHistories = question.delete(loginUser);
        deleteHistoryService.saveAll(deleteHistories);
```